memory/migrate_pages.py.data/node_move_pages.c

```diff
@@ -328,17 +330,26 @@ int main(int argc, char *argv[])
 			chunks = strtoul(optarg, NULL, 10);
                         break;
                 case 'o':
+#ifdef HAVE_HUGETLB_HEADER
 			overcommit = 1;
-                        break;
-                case 't':
+#else			
+                        overcommit = 0;
+#endif			
+			break;
+
+		case 't':
 			thp = 1;
                         break;
                 case 'h':
+#ifdef HAVE_HUGETLB_HEADER
 			hugepage = 1;
 			pagesize = gethugepagesize();
 			/* Using 1% of system memory for hugepages*/
         		memory = (total_mem) / 100;
-                        break;
+#else
+			hugepage = 0 ;
```

Comment:
Same here. You can move the case o: stanza and then club it with case h: inside a #ifdef #endif 
-                case 'o':
-                       overcommit = 1;
-                        break;
                 case 't':
                        thp = 1;
                         break;
+#ifdef HAVE_HUGETLB_HDR
+                case 'o':
+                       overcommit = 1;
+                        break;
                 case 'h':
                        hugepage = 1;
                        pagesize = gethugepagesize();
                        /* Using 1% of system memory for hugepages*/
                        memory = (total_mem) / 100;
                         break;
+#endif

Reply:
Done !